pkg/cluster/px/px.go

```diff
@@ -626,9 +627,14 @@ func (ops *pxClusterOps) upgradePX(spec apiv1beta1.ClusterSpec) error {
 			return err
 		}
 
-		daemonsetReadyTimeout, err := ops.getDaemonSetReadyTimeout()
-		if err != nil {
-			return err
+		var daemonsetReadyTimeout time.Duration
+		if opts.TalismanTimeout == 0 {
+			daemonsetReadyTimeout, err = ops.getDaemonSetReadyTimeout()
+			if err != nil {
+				return err
+			}
+		} else {
+			daemonsetReadyTimeout = time.Duration(opts.TalismanTimeout) * time.Minute
```

Comment:
the timeout given by the user should not be a daemonset timeout (i.e not be a cluster scoped timeout) but rather a per node timeout.